File: frontend/lib/models/pagination.dart
class Pagination {
  final int totalItems;
  final int totalPages;
  final int currentPage;
  final int pageSize;

  Pagination({
    required this.totalItems,
    required this.totalPages,
    required this.currentPage,
    required this.pageSize,
  });

  factory Pagination.fromJson(Map<String, dynamic> json) {
    return Pagination(
      totalItems: json['total_items'],
      totalPages: json['total_pages'],
      currentPage: json['current_page'],
      pageSize: json['page_size'],
    );
  }
}

File: forge/forge/agent_protocol/models/pagination.py
from pydantic import BaseModel, Field


class Pagination(BaseModel):
    total_items: int = Field(description="Total number of items.", examples=[42])
    total_pages: int = Field(description="Total number of pages.", examples=[97])
    current_page: int = Field(description="Current_page page number.", examples=[1])
    page_size: int = Field(description="Number of items per page.", examples=[25])
